app/uk/gov/hmrc/agentsubscriptionfrontend/service/SubscriptionService.scala

```diff
@@ -20,17 +20,38 @@ import javax.inject.{Inject, Singleton}
 
 import play.api.Logger
 import play.api.http.Status
-import uk.gov.hmrc.agentmtdidentifiers.model.Arn
+import uk.gov.hmrc.agentmtdidentifiers.model.{Arn, Utr}
 import uk.gov.hmrc.agentsubscriptionfrontend.connectors.AgentSubscriptionConnector
 import uk.gov.hmrc.agentsubscriptionfrontend.controllers.SubscriptionDetails
 import uk.gov.hmrc.agentsubscriptionfrontend.models._
 import uk.gov.hmrc.http.{HeaderCarrier, Upstream4xxResponse}
 
 import scala.concurrent.{ExecutionContext, Future}
 
+case class SubscriptionReturnedHttpError(httpStatusCode: Int) extends Product with Serializable
+
+object SubscriptionState extends Enumeration {
+  type SubscriptionState = Value
+  val BrandNewSubscription, IsOnlySubscribedInETMP, IsSubscribedToAgentServices, NoRegistrationFound = Value
```

Comment:
What does "NoRegistrationFound" mean? Is it " no subscription found in etmp"?

Reply:
yes, getSubscriptionStatus = NoRegistrationFound     in ETMP,
I can rename